src/base/permutationsutil.h

```diff
@@ -0,0 +1,79 @@
+/*
+   Copyright 2016 Software Reliability Lab, ETH Zurich
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+
+#ifndef BASE_PERMUTATIONSUTIL_H_
+#define BASE_PERMUTATIONSUTIL_H_
+
+#include <limits>
+
+void ComputeAllPermutations(std::vector<int> v,
```

Comment:
permutations_beam_size is not an obvious parameter of this function. Write a comment for it.